docs/data-cookbooks/tracking-files.md

```diff
@@ -0,0 +1,121 @@
+Cumulus has a number of options for tracking files and publishing files to CMR under certain metadata elements or with specified file types.
+We will cover Cumulus file types, collection configuration, and effects on publishing to CMR.
+
+### File types
+
+Cumulus follows the Cloud Notification Mechanism (CNM) file type conventions. Under this schema, there are four data types:
+
+* `data`
+* `browse`
+* `metadata`
+* `qa`
+
+In Cumulus, these data types are mapped to the `Type` attribute on `RelatedURL`s for UMM-G metadata, or used to map
+resources to one of `OnlineAccessURL`, `OnlineResource` or `AssociatedBrowseImages` for ECHO10 XML metadata.
+
+### Collection Configuration
+
+File types for each file in a granule can be configured at the collection level as below:
+
+```json
+{
+  "name": "MOD09GQ",
+  "version": "006",
+  "dataType": "MOD09GQ",
+  "granuleId": "^MOD09GQ\\.A[\\d]{7}\\.[\\S]{6}\\.006\\.[\\d]{13}$",
+  "granuleIdExtraction": "(MOD09GQ\\..*)(\\.hdf|\\.cmr|_ndvi\\.jpg)",
+  "sampleFileName": "MOD09GQ.A2017025.h21v00.006.2017034065104.hdf",
+  "files": [
+    {
+      "bucket": "protected",
+      "regex": "^MOD09GQ\\.A[\\d]{7}\\.[\\S]{6}\\.006\\.[\\d]{13}\\.hdf$",
+      "sampleFileName": "MOD09GQ.A2017025.h21v00.006.2017034065104.hdf",
+      "fileType": "data"
+    },
+    {
+      "bucket": "protected-2",
+      "regex": "^MOD09GQ\\.A[\\d]{7}\\.[\\S]{6}\\.006\\.[\\d]{13}\\.cmr\\.xml$",
+      "sampleFileName": "MOD09GQ.A2017025.h21v00.006.2017034065104.cmr.xml",
+      "fileType": "metadata"
+    },
+    {
+      "bucket": "public",
+      "regex": "^MOD09GQ\\.A[\\d]{7}\\.[\\S]{6}\\.006\\.[\\d]{13}_ndvi\\.jpg$",
+      "sampleFileName": "MOD09GQ.A2017025.h21v00.006.2017034065104_ndvi.jpg",
+      "fileType": "browse"
+    }
+  ]
+}
+```
+
+Files on a particular granule which match a given file specification will be receive the provided `fileType` in the granule record,
+which will then be used to inform CMR metadata updates in the publish step.
+
+### Publish to CMR
+
+When publishing granule metadata to CMR, the `PostToCmr` task will perform a few metadata updates in the area of URLs for its respective metadata schema.
+The table below shows how the CNM data types map to CMR Metadata updates.
+The UMM-G column reflects the `RelatedURL`'s `Type` derived from the CNM type, whereas the ECHO10 column shows how the CNM type affects the destination element.
+
+|CNM Type |UMM-G Location |ECHO10 Location |
+| ------  | ------ | ------ |
+| `data` | `RelatedURL.Type = 'GET DATA'` | `OnlineAccessURL` |
+| `browse` | `RelatedURL.Type = 'GET RELATED VISUALIZATION'` | `AssociatedBrowseImage` |
+| `metadata` | `RelatedURL.Type = 'EXTENDED METADATA` | `OnlineResource` |
+| `qa` | `RelatedURL.Type = 'EXTENDED METADATA'` | `OnlineResource` |
+
+### Common Use Cases
+
+This section briefly documents some common use cases and the recommended collection configuration for the file.
+
+Configuring browse imagery:
```

Comment:
This may be misleading to end users - this configuration only maps these types if used in a workflow with discovery or another lambda that utilizes this value to update granules.    

For the moment (to my knowledge) none of the core lambdas outside of DiscoverGranules support it, and we're not currently explicitly promising consistency in this usage.      The granule file fileType is explicitly intended to be supported, the method of setting that may vary.   This value *can* be used to support that, but for the moment ParsePDR and the CNM workflows ignore it entirely in favor of whatever is in the ingest message.

Reply:
Noted, I'll change up that section with the relevant information.

Reply:
7af13a26a55c98b8df549842e36bdbfda48ca3c6